components/lib/dataprotect/src/main/java/mozilla/components/lib/dataprotect/Keystore.kt

```diff
@@ -82,15 +94,24 @@ open class KeyStoreWrapper {
      * @throws NoSuchAlgorithmException If the cipher algorithm is not supported
      */
     open fun makeKeyFor(label: String): SecretKey {
-        val spec = KeyGenParameterSpec.Builder(label,
-                KeyProperties.PURPOSE_ENCRYPT or KeyProperties.PURPOSE_DECRYPT)
+        if (Build.VERSION.SDK_INT >= M) {
+            val spec = KeyGenParameterSpec.Builder(
+                label,
+                PURPOSE_ENCRYPT or PURPOSE_DECRYPT
+            )
                 .setKeySize(CIPHER_KEY_LEN)
                 .setBlockModes(CIPHER_MOD)
                 .setEncryptionPaddings(CIPHER_PAD)
                 .build()
-        val gen = KeyGenerator.getInstance(CIPHER_ALG, KEYSTORE_TYPE)
-        gen.init(spec)
-        return gen.generateKey()
+            val gen = KeyGenerator.getInstance(CIPHER_ALG, KEYSTORE_TYPE)
+            gen.init(spec)
+            return gen.generateKey()
+        } else {
+            val spec = SecureRandom()
+            val gen = KeyGenerator.getInstance(CIPHER_ALG, KEYSTORE_TYPE)
+            gen.init(spec)
```

Comment:
You probably want to use `init (AlgorithmParameterSpec params, SecureRandom random)` and specify key size, block mode, encryption padding.